### DISCUSSION TITLE
Delete All Measures.

### DISCUSSION THREAD

**Discussion by SharmaAntriksh** [TabularEditor/TabularEditor3#896] [Other Topics]
I am trying to delete every measure using the Delete Method however I don't understand why the below code isn't supported.

```
foreach (Table t in Model.Tables)
{
    foreach (Measure m in t.Measures)
    {
        m.Delete();
    }
}
```

It throws an error that System.InvalidOperationException: Collection was modified; enumeration operation may not execute.

For now I am using an alternate way of loading table name and measure name in a list and then running the Model.Tables.Measure.Delete method

```
var measureNames = new List<(string tableName, string measureName)>();

foreach(Measure m in Model.AllMeasures)
{
    measureNames.Add(
        (m.Table.Name, m.Name)
    );
}

measureNames.ForEach(
    (row) => 
    Model.Tables[row.tableName].Measures[row.measureName].Delete()
);
```

**Comment by otykier** [comment]
This is standard in .NET - you can't enumerate a collection that is being modified. Copying the objects you intend to enumerate into a list, is indeed the correct way of solving this. You can make it more compact by writing this:

```csharp
foreach (Table t in Model.Tables)
{
    foreach (Measure m in t.Measures.ToList())
    {
        m.Delete();
    }
}
```
